From c820b92baad8444fc01b29964db11e1897ff156f Mon Sep 17 00:00:00 2001
From: Peng Ma <peng.ma@nxp.com>
Date: Wed, 12 Sep 2018 19:01:23 +0800
Subject: [PATCH 059/741] dmaengine: fsl-qdma: Defer probing if no MC portal
 available

commit 80c2865b2c79a1222c458b0a047aca3660b6598d from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

MC portals may not be available at the initial probing attempt
due to dependencies on other modules.

Signed-off-by: Peng Ma <peng.ma@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/dma/dpaa2-qdma/dpaa2-qdma.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/dma/dpaa2-qdma/dpaa2-qdma.c b/drivers/dma/dpaa2-qdma/dpaa2-qdma.c
index 975918b11269..c29eda063468 100644
--- a/drivers/dma/dpaa2-qdma/dpaa2-qdma.c
+++ b/drivers/dma/dpaa2-qdma/dpaa2-qdma.c
@@ -789,7 +789,10 @@ static int dpaa2_qdma_probe(struct fsl_mc_device *dpdmai_dev)
 	/* obtain a MC portal */
 	err = fsl_mc_portal_allocate(dpdmai_dev, 0, &priv->mc_io);
 	if (err) {
-		dev_err(dev, "MC portal allocation failed\n");
+		if (err == -ENXIO)
+			err = -EPROBE_DEFER;
+		else
+			dev_err(dev, "MC portal allocation failed\n");
 		goto err_mcportal;
 	}
 
-- 
2.17.1

